src/VisualStudio/LiveShare/Impl/Shims/GoToDefinitionHandlerShim.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LiveShare.LanguageServices;
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare.Shims
+{
+    internal class GoToDefinitionHandlerShim : AbstractLiveShareHandlerShim<TextDocumentPositionParams, object>
+    {
+        public GoToDefinitionHandlerShim(IEnumerable<Lazy<IRequestHandler, IRequestHandlerMetadata>> requestHandlers)
+            : base(requestHandlers, Methods.TextDocumentDefinitionName)
+        {
+        }
+    }
+
+    [ExportLspRequestHandler(LiveShareConstants.CSharpContractName, Methods.TextDocumentDefinitionName)]
+    internal class CSharpGoToDefinitionHandlerShim : GoToDefinitionHandlerShim
+    {
+        [ImportingConstructor]
+        public CSharpGoToDefinitionHandlerShim([ImportMany] IEnumerable<Lazy<IRequestHandler, IRequestHandlerMetadata>> requestHandlers)
+            : base(requestHandlers)
+        {
+        }
+    }
+
+    [ExportLspRequestHandler(LiveShareConstants.VisualBasicContractName, Methods.TextDocumentDefinitionName)]
+    internal class VisualBasicGoToDefinitionHandlerShim : GoToDefinitionHandlerShim
```

Comment:
why not just have one and export if for both C# and VB?

Reply:
compiler error - duplicate attribute